### PR TITLE
ci: bump release build to ubuntu-22.04

### PR DESCRIPTION
**Pull request by jwnrt**
Required to get the necessary `glib` version required for QEMU (>=2.66.0).

Note that binaries built from `ubuntu-22.04` will no longer be usable from `ubuntu-20.04` due to `glibc` incompatibility.

### CODE REVIEW

**Review by rivos-eblot — APPROVED** [review]

**Comment by jwnrt** [comment]
Tested the workflow on my fork and it seems to work fine:
https://github.com/jwnrt/qemu/actions/runs/11591874700

@rivos-eblot do you think we could cut a release of `ot-darjeeling-9.1.0` once this is merged? It would help with restarting Bazel integration work in the opentitan repo

File: scripts/opentitan/make_release.sh [hunk 1]
```diff
@@ -18,4 +18,4 @@ QEMU_BUILD="$3"
 tar --create --auto-compress --verbose --file="$OUT_TARBALL" \
     --directory="$QEMU_DIR" \
     "$QEMU_BUILD"/qemu-{system-riscv32,img} \
-    scripts/opentitan/{otpconv,flashgen}.py
+    scripts/opentitan/{otptool,flashgen}.py
```
Comment (rivos-eblot) [inline scripts/opentitan/make_release.sh]: `otptool.py` depends on many other Python modules which are located in `python/qemu/ot...`.
I'm not the author of this script, I'm not sure about its purposes but is it ok to put the script into the tarball but ignore its dependencies?

**Comment by jwnrt** [comment]
And should I prefix these commits `[ot]`?

**Review by rivos-eblot — COMMENTED** [review]
otptool.py dependencies might be missing

File: scripts/opentitan/make_release.sh (hunk 1 above)
Comment (jwnrt) [reply scripts/opentitan/make_release.sh]: Good catch, I've included that directory

**Comment by rivos-eblot** [comment]
> And should I prefix these commits `[ot]`?

Yes you are right, as well as the directory (see comment from #80)

**Comment by rivos-eblot** [comment]
> @rivos-eblot do you think we could cut a release of `ot-darjeeling-9.1.0` once this is merged? It would help with restarting Bazel integration work in the opentitan repo

I need to push a couple of fixes first, but yes, by the end of the day this should be ok.

**Pull request merged**